src/dotnet-install.ps1

```diff
@@ -204,13 +204,28 @@ function Get-Machine-Architecture() {
         return $ENV:PROCESSOR_ARCHITEW6432
     }
 
-    # covers the case when PS x64 is run on ARM machine
-    if( ((get-wmiobject Win32_OperatingSystem).OSArchitecture) -like "ARM*") {
-        if( [Environment]::Is64BitOperatingSystem )
-        {
-            return "arm64"
-        }  
-        return "arm"
+    # Not all platforms have Get-CimInstance (preferred) or Get-WmiObject.
+    # It shouldn't be fatal if neither of those cmdlets exists, however
+    # some NanoServer/PS images contain a non-functioning version of Get-CimInstance
+    # so don't even try it if we're on one of those images.
+
+    $IsNanoServer = $false
+    try {
+        $IsNanoServer = (Get-ItemPropertyValue 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion\Server\ServerLevels' -Name NanoServer -ErrorAction Ignore) -eq 1
```

Comment:
I recommend not to care whether the machine is a nano server or something else.
Today, nano server doesn't have Get-CimInstance. But maybe next version will. Maybe someone installed these cmdlets manually onto their machine.

I suggest the following:
```powershell
try {
    Get-CimInstance .....
}
catch {
    # Machine doesn't support Get-CimInstance
}

# fallback to old way
return $ENV:PROCESSOR_ARCHITECTURE
```